src/main/java/edu/ksu/canvas/attendance/controller/SummaryController.java

```diff
@@ -78,47 +83,89 @@ private ModelAndView studentSummary(String sectionId, String studentId, boolean
-                            new AttendanceSummaryModel.Entry(entry.getCourseId(), entry.getSectionId(), entry.getStudentId(), entry.getSisUserId(), entry.getStudentName(), student.getDeleted(), entry.getTotalClassesTardy(), entry.getTotalClassesMissed(), entry.getTotalClassesExcused(), entry.getTotalClassesPresent())
-                          : new AttendanceSummaryModel.Entry(entry.getCourseId(), entry.getSectionId(), entry.getStudentId(), entry.getSisUserId(), entry.getStudentName(), student.getDeleted(), entry.getSumMinutesMadeup(), entry.getRemainingMinutesMadeup(), entry.getSumMinutesMissed(), entry.getPercentCourseMissed())));
+                .filter(entry -> entry.getStudentId() == validatedStudentId);
+
+        summaryForSections.addAll(isSimpleAttendance ?
+                reportService.getSimpleAttendanceSummaryReport(student.getCanvasSectionId()) :
+                reportService.getAviationAttendanceSummaryReport(student.getCanvasSectionId()));
+
+        List<LtiLaunchData.InstitutionRole> institutionRoles = canvasService.getRoles();
 
         institutionRoles.stream()
                 .filter(institutionRole -> institutionRole.equals(LtiLaunchData.InstitutionRole.Learner))
                 .findFirst()
                 .ifPresent(role -> page.addObject("isStudent", true));
 
+        int totalTardy = 0, totalExcused = 0, totalMissed = 0;
+
+        if (!studentAttendanceList.isEmpty()) {
+            for (int i = 0; i < studentAttendanceList.size(); i++) {
+                AttendanceStudent current = studentAttendanceList.get(i);
+
+                for (int index = 0; index < current.getAttendances().size(); index++) {
+                    switch (current.getAttendances().get(index).getStatus()) {
+                        case TARDY:
+                            totalTardy++;
+                            break;
+                        case EXCUSED:
+                            totalExcused++;
+                            break;
+                        case ABSENT:
+                            totalMissed++;
+                            break;
+
+                    }
+                }
+            }
+        }
 
-        page.addObject("sectionId", sectionId);
+        page.addObject("totalTardy", totalTardy);
+        page.addObject("totalExcused", totalExcused);
+        page.addObject("totalMissed", totalMissed);
+        page.addObject("selectedSectionId", sectionId);
+        page.addObject("sectionList", sectionList);
         page.addObject("student", student);
+        page.addObject("studentList", studentAttendanceList);
         page.addObject("summaryForm", makeupForm);
+        page.addObject("dropDownList", DropDownOrganizer.sortWithSelectedSectionFirst(sectionList, sectionId));
 
         return page;
     }
 
+    public void getAttendanceTotals(AttendanceStudent entry, int totalTardy, int totalExcused, int totalMissed){
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "entry". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1172)